### PR TITLE
Introduces input macros

### PR DESCRIPTION
**Pull request by zonble**
The PR introduces input macros. Input macros are candidates that would be
replace dynamically by the input engine. We use input macros to input date and
time since a user might be hardly to remember the current date (especially in
the traditional Chinese calendar such as 癸卯年).

Currently, it supports the following input macros:

| 輸入     | 範例                    |
| -------- | ----------------------- |
| 今天日期 | 2023 年 12 月 11 日     |
| 明天日期 | 民國 112 年 12 月 12 日 |
| 現在時間 | 下午 2:28:21            |
| 所在時區 | 台北標準時間            |
| 今年干支 | 癸卯年                  |
| 今年生肖 | 水兔年                  |

The localized dates are powered by ICU, so ICE becomes a dependency.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,12 @@ sudo apt install \
     cmake extra-cmake-modules gettext libfmt-dev
```
Comment (lukhnos) [inline README.md:35]: Nit: should we just add `libicu-dev` here? Since we are already telling people they need to install those `-dev` packages here…

**Review by lukhnos — CHANGES_REQUESTED** [review]
Would you like to cherry-pick https://github.com/openvanilla/fcitx5-mcbopomofo/commit/40f78ba587d2844fbf19c35c077ab92c0026174e , which uses an `std::unordered_map` for the macro lookup?

Otherwise LGTM!

**Comment by zonble** [comment]
> Would you like to cherry-pick [40f78ba](https://github.com/openvanilla/fcitx5-mcbopomofo/commit/40f78ba587d2844fbf19c35c077ab92c0026174e) , which uses an `std::unordered_map` for the macro lookup?
> 
> Otherwise LGTM!

The branch was merged.

File: README.md (hunk 1 above)
Comment (zonble) [reply README.md]: Updated at 84b0dc5053ac9795ec105b91933e126606105e43

**Review by lukhnos — APPROVED** [review]

File: build.sh [hunk 2]
```diff
@@ -4,5 +4,5 @@ rm -rf build
 mkdir -p build
 cd build
 cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Debug # use Debug for easy debugging with gdb
-make # or ninja, depending on your system
-sudo make install # or sudo ninja install
+make                                                          # or ninja, depending on your system
```
Comment (xatier) [inline build.sh:7]: nit: to use `ninja` we need to add `-G Ninja` in `cmake` command. @lukhnos would you like to create a small PR for this?

https://github.com/openvanilla/fcitx5-mcbopomofo/issues/96

**Pull request merged**